libraries/AP_HAL/CAN.h

```diff
@@ -0,0 +1,128 @@
+                    unchanged.
+
+        return      0 - CAN port open failed
+                        code of error can be read with get_last_error() method
+                    1 - CAN port open succeeded
+     */
+    virtual bool begin(uint32_t bitrate) = 0;
+
+    /*  CAN port extended open method
+
+        Opens port with specified bit rate and type.
+        CAN 2.0B is back compatible with CAN 2.0A and such devices can be opened with both types
+
+        bitrate     Selects the speed that the port will be configured to.  If zero, the port speed is left
+                    unchanged.
+        samplepoint Sample point of bit in percents.
+                    Like the bit rate, the value of the sample point is usually selected by an industry standards
+                    group or by the systems engineering section of the company creating the distributed product.
+                    Contrary to the 50% sample point of the common serial port or UART, (Universal Asynchronous Receiver
+                    Transmitter), the structure of the CAN protocol places the optimal sample point much closer to the end
+                    of the bit period. Sample points in the high 80% range are required for industry standards
+                    J1939 (250K BPS) and J2284 (500K BPS). For UAVCAN recommended value is 87.5%.
+
+        return      0 - CAN port open failed
+                        code of error can be read with get_last_error() method
+                    1 - CAN port open succeeded
+     */
+    virtual bool begin(uint32_t bitrate, float samplepoint) = 0;
+
+    /*
+        CAN port close
+
+        return      0 - CAN port close failed
+                        code of error can be read with get_last_error() method
+                    1 - CAN port close succeeded
+     */
+    virtual void end() = 0;
+
+    /*
+        Reset opened CAN port
+
+        Pending messages to be transmitted are deleted and receive state and FIFO also reset.
+        All pending errors are cleared.
+     */
+    virtual void reset() = 0;
+
+    /*
+        Test if CAN port is opened and initialized
+
+        return      0 - CAN port not initialized
+                        code of error can be read with get_last_error() method
+                    1 - CAN port is initialized
+     */
+    virtual bool is_initialized() = 0;
+
+    /*
+        Return if CAN port has some untransmitted pending messages
+
+        return     -1 - CAN port is not opened or initialized
+                    0 - no messages are pending
+             positive - number of pending messages
+     */
+    virtual int32_t tx_pending() = 0;
+
+    /*
+        Return if CAN port has received but yet read messages
+
+        return     -1 - CAN port is not opened or initialized
+                    0 - no messages are pending for read
+             positive - number of pending messages for read
+     */
+    virtual uint32_t available() = 0;
```

Comment:
I agree, this was taken from other existing HAL drivers. I will consider to remove it.